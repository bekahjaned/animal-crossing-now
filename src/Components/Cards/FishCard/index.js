import React from "react";
import "../index.css";

class FishCard extends React.Component {
  render() {
    return (
      <div className="critter">
        <img
          src={require(`../../../data/images/Fish/${this.props.image}.png`)}
          alt="fish"
        ></img>
        <p className="critter-name">{this.props.name}</p>
        <p>{this.props.location}</p>
      </div>
    );
  }
}

export default FishCard;
