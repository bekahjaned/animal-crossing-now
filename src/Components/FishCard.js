import React from "react";

class FishCard extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="critter">
        <img
          src={require(`../data/images/Fish/${this.props.image}.png`)}
          alt="fish"
        ></img>
        <h3>{this.props.name}</h3>
        <p>{this.props.location}</p>
      </div>
    );
  }
}

export default FishCard;
