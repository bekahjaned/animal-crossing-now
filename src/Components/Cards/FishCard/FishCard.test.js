import React from "react";
import renderer from "react-test-renderer";

import FishCard from "./index";

it("renders a FishCard", () => {
  // given
  const image = "Squid";
  const name = "Squid";
  const location = "Sea";
  const availability = "0 - 23";
  const price = "500";

  // when
  const tree = renderer
    .create(
      <FishCard
        image={image}
        name={name}
        location={location}
        availability={availability}
        price={price}
      />
    )
    .toJSON();

  // then
  expect(tree).toMatchSnapshot();
});
