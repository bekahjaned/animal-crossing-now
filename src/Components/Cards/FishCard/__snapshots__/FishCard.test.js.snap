// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a FishCard 1`] = `
<div
  className="critter"
>
  <img
    alt="fish"
    src="Squid.png"
  />
  <p
    className="critter-name"
  >
    Squid
  </p>
  <p>
    Sea
  </p>
  <p
    className="availability"
  >
    Available: 
    all day
  </p>
  <p>
    Price: 
    500
     Bells
  </p>
</div>
`;
